discovery-provider/src/queries/get_attestation.py

```diff
@@ -61,10 +61,27 @@ class AttestationError(Exception):
     pass
 
 
+eth_abi_values = load_eth_abi_values()
+REWARDS_CONTRACT_ABI = eth_abi_values["EthRewardsManager"]["abi"]
+
+eth_web3 = Web3(HTTPProvider(shared_config["web3"]["eth_provider_url"]))
+eth_registry_address = eth_web3.toChecksumAddress(
+    shared_config["eth_contracts"]["registry"]
+)
+eth_registry_instance = eth_web3.eth.contract(
+    address=eth_registry_address, abi=eth_abi_values["Registry"]["abi"]
+)
+eth_rewards_manager_address = eth_registry_instance.functions.getContract(
+    bytes("EthRewardsManagerProxy", "utf-8")
+).call()
+
+
 def is_valid_oracle(address: str):
-    # TODO: flesh this out, refresh oracle addresses [AUD-729]
-    default_oracle_address = shared_config["discprov"]["default_oracle_address"]
-    return address == default_oracle_address
+    eth_rewards_manager_instance = eth_web3.eth.contract(
+        address=eth_rewards_manager_address, abi=REWARDS_CONTRACT_ABI
+    )
+    oracle_addresses = eth_rewards_manager_instance.functions.getAntiAbuseOracleAddresses().call()
+    return address in oracle_addresses
```

Comment:
I would probably do what we've done in node.js infra before and do this on some set interval rather than on-demand. I imagine this function will get called *a lot*

Reply:
since we don't have the super easy setTimeout way here, you could look into just caching this in redis with a TTL.
you may want to just look into the monitors infra for it too -- that would give you what you want for almost free https://github.com/AudiusProject/audius-protocol/blob/master/discovery-provider/src/monitors/monitors.py#L41

Reply:
Made the change using threading.Timer. Then changed the change in favor of a celery task. Check out last couple of commits. 
Maybe instead of a redis_set_and_dump and redis_get_or_restore, we could just do a simple redis.set and redis.get, and if the key is not there then we make the call to chain. This way there is no file creation/update. wdyt?

Reply:
Ah. made the changes before seeing your last comment